src/robotFactory.js

```diff
@@ -1,10 +1,80 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    // write code here
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    if (coords && Object.keys(coords).length) {
+      this.coords = coords;
+    } else {
+      this.coords = {
+        x: 0, y: 0,
+      };
+    }
+    this.chipVersion = chipVersion;
+  }
+  getInfo() {
+    return `Robot: ${this.name}, `
+    + `Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  };
+}
 
-class DeliveryDrone {}
+BaseRobot.prototype.goForward = function(step = 1) {
+  this.coords.y += step;
+};
+
+BaseRobot.prototype.goBack = function(step = 1) {
+  this.coords.y -= step;
+};
+
+BaseRobot.prototype.goLeft = function(step = 1) {
+  this.coords.x -= step;
+};
+
+BaseRobot.prototype.goRight = function(step = 1) {
+  this.coords.x += step;
+};
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    const coordsZ = {
+      z: 0,
+      ...coords,
+    };
+
+    super(name, weight, coordsZ, chipVersion);
+  }
```

Comment:
why do you send coord Z to Base robot? It don't need in this coordinate
```suggestion
    super(name, weight, coords, chipVersion);
  }
```